### PR TITLE
Idle-tuning is enabled by default in 0.12.0

### PR DESCRIPTION
**Pull request by SueChaplain**
GC on idle and compact on idle are now enabled
by default when OpenJ9 detects that it is running
in a container and the VM enters an idle state.

Topics udpated to introduce the new behaviour.

Closes: #112

Signed-off-by: Sue Chaplain <sue_chaplain@uk.ibm.com>

### CODE REVIEW

**Comment by SueChaplain** [comment]
@ashu-mehra

Hi. Please can you review the changes for this update? If you want to see a draft of the built doc, please let me know and I will share a private build.

File: docs/xxidletuningcompactonidle.md [hunk 1]
```diff
@@ -26,23 +26,23 @@
 
 **(Linux&reg; only)**
 
-This option controls garbage collection processing with compaction when the status of the OpenJ9 VM is set to idle.
+This option controls garbage collection processing with compaction when the OpenJ9 VM is running in a container and the state of the VM is set to idle.
```
Comment (ashu-mehra) [inline docs/xxidletuningcompactonidle.md]: To me this update gives an impression that the option is applicable only when OpenJ9 is running in container, which is not correct. I feel this does not need any change.

File: docs/xxidletuningcompactonidle.md [hunk 2]
```diff
@@ -26,23 +26,23 @@
...
 |--------------------------------|---------|:----------------------------------------------------------------------------------:|
-| `-XX:+IdleTuningCompactOnIdle` | Enable  |                                                                                    |
-| `-XX:-IdleTuningCompactOnIdle` | Disable | <i class="fa fa-check" aria-hidden="true"></i><span class="sr-only">yes</span> |
+| `-XX:+IdleTuningCompactOnIdle` | Enable  |  <i class="fa fa-check" aria-hidden="true"></i><span class="sr-only">yes</span>    |
+| `-XX:-IdleTuningCompactOnIdle` | Disable |                                                                                    |
 
 
-If you enable this option, the OpenJ9 VM attempts to compact the object heap when the VM status is idle.
+If the OpenJ9 VM is running in a container and the state is set to idle, the VM attempts to compact the object heap following a garbage collection cycle. The garbage collection cycle is controlled by the `-XX:+IdleTuningGcOnIdle` option, which is also enabled by default when the OpenJ9 VM is running inside a container.
```
Comment (ashu-mehra) [inline docs/xxidletuningcompactonidle.md]: Again, this behavior is not specific to container env, so the statement `If the OpenJ9 VM is running in a container` can be avoided.

File: docs/xxidletuningcompactonidle.md [hunk 3]
```diff
@@ -26,23 +26,23 @@
...
 
 ## Syntax
 
         -XX:[+|-]IdleTuningCompactOnIdle
 
 | Setting                        | Effect  | Default                                                                            |
 |--------------------------------|---------|:----------------------------------------------------------------------------------:|
-| `-XX:+IdleTuningCompactOnIdle` | Enable  |                                                                                    |
-| `-XX:-IdleTuningCompactOnIdle` | Disable | <i class="fa fa-check" aria-hidden="true"></i><span class="sr-only">yes</span> |
+| `-XX:+IdleTuningCompactOnIdle` | Enable  |  <i class="fa fa-check" aria-hidden="true"></i><span class="sr-only">yes</span>    |
```
Comment (ashu-mehra) [inline docs/xxidletuningcompactonidle.md]: The `Default` depends on whether we are running in container or not.
When not in container, default is `-XX:-IdleTuningCompactOnIdle`.
When in container, default is `-XX:+IdleTuninigCompactOnIdle`.
Probably we need two columns - one for non-container case and another for container case.

File: docs/xxidletuninggconidle.md [hunk 4]
```diff
@@ -26,29 +26,29 @@
...
 
-This option is used with `-XX:IdleTuningMinIdleWaitTime` and `-XX:IdleTuningMinFreeHeapOnIdle` . If values for these options are not explicitly specified, the VM sets the following defaults:
+This option is used with the `-XX:IdleTuningMinIdleWaitTime` and `-XX:IdleTuningMinFreeHeapOnIdle` options. If values for these options are not explicitly specified, the VM sets the following defaults:
 
 -   `-XX:IdleTuningMinIdleWaitTime`=180
 -   `-XX:IdleTuningMinFreeHeapOnIdle`=0
 
 
-If you want the VM to attempt to compact the object heap before releasing memory, use the`-XX:+IdleTuningCompactOnIdle` option.
+By default, the VM also attempts to compact the object heap before releasing memory, which is controlled by the `-XX:[+|-]IdleTuningCompactOnIdle` option.
```
Comment (ashu-mehra) [inline docs/xxidletuninggconidle.md]: This change in default behavior is applicable only when JVM is running in container. I think that is not evident here.

File: docs/xxidletuninggconidle.md [hunk 5]
```diff
@@ -26,29 +26,29 @@
...
 
 ## Syntax
 
         -XX:[+|-]IdleTuningGcOnIdle
 
 | Setting                   | Effect  | Default                                                                            |
 |---------------------------|---------|:----------------------------------------------------------------------------------:|
-| `-XX:+IdleTuningGcOnIdle` | Enable  |                                                                                    |
-| `-XX:-IdleTuningGcOnIdle` | Disable | <i class="fa fa-check" aria-hidden="true"></i><span class="sr-only">yes</span> |
+| `-XX:+IdleTuningGcOnIdle` | Enable  | <i class="fa fa-check" aria-hidden="true"></i><span class="sr-only">yes</span>     |
```
Comment (ashu-mehra) [inline docs/xxidletuninggconidle.md]: Same comment as for `xxidletuningcompactonidle` doc page.

File: docs/xxidletuninggconidle.md [hunk 6]
```diff
@@ -26,29 +26,29 @@
 
 **(Linux&reg; only)**
 
-This option can be used to reduce the memory footprint of the OpenJ9 VM when it is in an idle state.
+When the OpenJ9 VM is running in a container, this option controls whether a garbage collection cycle takes place when the state of the OpenJ9 VM is set to idle.
```
Comment (ashu-mehra) [inline docs/xxidletuninggconidle.md]: Same comment as for `xxidletuningcompactonidle` doc page.

File: docs/xxidletuningminidlewaittime.md [hunk 7]
```diff
@@ -36,7 +36,7 @@ When the OpenJ9 VM is idle, this option controls the minimum length of time that
 
 | Setting     |  Value           | Default  |
 |-------------|------------------|----------|
-|`<secs>`     | *[0 or greater]* | 0        |
+|`<secs>`     | *[0 or greater]* | 180      |
```
Comment (ashu-mehra) [inline docs/xxidletuningminidlewaittime.md]: I think we need to mention two default values, one for non-container case (default is 0) and another for container case (default is 180).

File: docs/xxidletuningminidlewaittime.md [hunk 8]
```diff
@@ -26,7 +26,7 @@
 
 ** (Linux&reg; only) **
 
-When the OpenJ9 VM is idle, this option controls the minimum length of time that the VM must be idle before the status of the VM is set to idle. Further tuning options control the compaction of the object heap and the release of free memory pages, which reduces the footprint of the VM.
+The OpenJ9 VM can automatically detect when it is running inside a container. When the OpenJ9 VM is idle, this option controls the minimum length of time that the VM must be idle before the state of the VM is set to idle. When the state changes to idle, a garbage collection cycle runs, the object heap is compacted, and free memory pages are released back to the operating system, which reduces the footprint of the VM. Garbage collection and compaction are controlled by the `-XX:+IdleTuningGcOnIdle` and `-XX:+IdleTuningCompactOnIdle` options, which are enabled by default when the OpenJ9 VM is running inside a container.
```
Comment (ashu-mehra) [inline docs/xxidletuningminidlewaittime.md]: I think the description is good, but the first statement is not really needed, as container detection is not related to this option.

**Comment by SueChaplain** [comment]
@ashu-mehra - thanks for putting me straight. (Had misremembered what this one was about). Hopefully I've fixed up all the topics appropriately. If you see any further issues, please let me know.

File: docs/xxidletuninggconidle.md [hunk 9]
```diff
@@ -26,30 +26,30 @@
...
+| `-XX:+IdleTuningGcOnIdle` | Enable  |  | <i class="fa fa-check" aria-hidden="true"></i><span class="sr-only">yes</span>     |
+| `-XX:-IdleTuningGcOnIdle` | Disable |  <i class="fa fa-check" aria-hidden="true"></i><span class="sr-only">yes</span>   |   |
 
+The default depends on whether or not the OpenJ9 VM is running in a docker container. As indicated in the table, when the VM is running in a container and the state is set to idle, this option causes the VM to release free memory pages in the object heap without resizing the Java&trade; heap. The pages are reclaimed by the operating system, which reduces the physical memory footprint of the VM.
 
-When the VM enters the idle state, enabling this option causes the VM to release free memory pages in the object heap without resizing the Java&trade; heap. The pages are reclaimed by the operating system, which reduces the physical memory footprint of the VM.
+In a container environment, the VM also attempts to compact the object heap before releasing memory, which is controlled by the `-XX:[+|-]IdleTuningCompactOnIdle` option.
```
Comment (ashu-mehra) [inline docs/xxidletuninggconidle.md]: This statement is applicable for non-container environment as well, isn't it?

Comment (SueChaplain) [reply docs/xxidletuninggconidle.md]: Originally I had that sentence as part of the last paragraph. Moved it back and added some clarification.

**Comment by ashu-mehra** [comment]
Looks fine to me. Thanks!

**Pull request merged**